internal/k8s/reconciler/validator/gateway_test.go

```diff
@@ -531,7 +531,7 @@ func TestListenerValidate(t *testing.T) {
 		require.NoError(t, err)
 
 		condition := listenerState.Status.ResolvedRefs.Condition(0)
-		assert.Equal(t, status.ListenerConditionReasonInvalidCertificateRef, condition.Reason)
+		assert.Equal(t, status.ListenerConditionReasonRefNotPermitted, condition.Reason)
```

Comment:
If the test name for this is accurate ("unsupported certificate type"), I believe the proper reason here would still be `InvalidCertificateRef`